console/module/api/src/main/java/org/eclipse/kapua/app/console/module/api/client/util/KapuaSafeHtmlUtils.java

```diff
@@ -30,6 +30,12 @@ public static String htmlUnescape(String safeHtml) {
             return null;
         }
 
+        if (safeHtml.indexOf("&amp;") != -1) {
+            safeHtml = safeHtml.replace("&amp;", "&");
+        }
+        if (safeHtml.indexOf("amp;") != -1) {
```

Comment:
Wouldn't this change every occurrence of the string "amp" in every string?

Reply:
@Coduz To answer your question, this wouldn't change every occurrence of the string `amp` but  only `amp;`.
There are cases where we have one or multiple unnecessary instances of `amp;` string added, like here:
`&amp;amp;amp;amp;amp;amp;quot;string&amp;`
the sting `&amp;` gets displayed correctly as `&` but we are still have multiple `amp;` instances that prevent the complete "unescape" for all of the string characters, and we are left with something like this:
`&amp;amp;amp;amp;amp;quot;string&;`
The string is partially "unescaped" but not completely.
With this solution the unnecessary `amp;` is  removed, and the correct value displayed after only one `htmlUnescape` call.
I guess that there is also the case where the user intentionally enters `amp` as a part of the string. But what probability is it for it to be entered in combination with`;` ?
What do you think?

Reply:
@Coduz Is it ok to merge this?